src/pjs/pjs.proto

```diff
@@ -0,0 +1,184 @@
+}
+
+service API {
+  // Job API
+	
+  // CreateJob creates a new job.
+  // Child jobs can be created by setting the context field to the appropriate parent job context.
+  rpc CreateJob(CreateJobRequest) returns (CreateJobResponse) {}
+  // CancelJob cancels a job.
+  // Canceling a job transitions all of the associated QUEUED and PROCESSING jobs to the DONE state and sets their error codes to CANCELED. 
+  // This will terminate all ongoing processing associated with the job.
+  // Nothing will be deleted.
+  // A job can only be canceled with the parent job context.
+  rpc CancelJob(CancelJobRequest) returns (CancelJobResponse) {}
+  // DeleteJob deletes a job.
+  // DeleteJob first cancels the job, then deletes all of the metadata and filesets associated with the job.
+  // A job can only be deleted with the parent job context.
+  rpc DeleteJob(DeleteJobRequest) returns (DeleteJobResponse) {}
+  // ListJob returns a list of jobs and information about each job.
+  // The jobs returned in the list are the child jobs of the provided job.
+  // If no job is provided, the list is the child jobs of the provided job context.
+  // The provided job must be associated with the provided job context or a descendant of the job associated with the provided job context.
+  rpc ListJob(ListJobRequest) returns (stream JobInfo) {}
+  // WalkJob returns a list of jobs in a hierarchy and information about each job.
+  // Walking a job traverses the job hierarchy rooted at the provided job.
+  // The provided job must be associated with the provided job context or a descendant of the job associated with the provided job context.
+  rpc WalkJob(WalkJobRequest) returns (stream JobInfo) {}
+  // InspectJob returns detailed information about a job.
+  rpc InspectJob(InspectJobRequest) returns (JobInfoDetails) {}
+	
+  // Queue API
+	
+  // ProcessQueue should be called by workers to process jobs in a queue.
+  // The protocol is as follows:
+  //   Worker sends an initial request with the queue id.
+  //   For each job:
+  //     Server sends a response with a job context and the associated queue element.
+  //     Worker processes the job.
+  //     Worker sends a request with the job output or indicates that the job failed.
+  // This RPC should generally be run indefinitely. Workers will be scaled based on demand, so the expectation is that they should be processing queues while they are up.
+  // This RPC will be canceled by the server if the current job is canceled. Workers should generally retry the RPC when disconnects occur.
+  rpc ProcessQueue(stream ProcessQueueRequest) returns (stream ProcessQueueResponse) {}
+  // ListQueue returns a list of queues and information about each queue.
+  rpc ListQueue(ListQueueRequest) returns (stream QueueInfo) {}
+  // InspectQueue returns detailed information about a queue.
+  rpc InspectQueue(InspectQueueRequest) returns (QueueInfoDetails) {}
+}
+
+// Job Messages
+
+message CreateJobRequest {
+  string context = 1;
```

Comment:
Even after reading the doc, I'm not sure what a context is.  It says it identifies the parent of this job, but parents of jobs are identified as Job objects elsewhere.  Why a string and how does a caller produce a correct string?

Reply:
This is a bearer token used to create a job from within another job.

Reply:
Could this info be included as comment(s) in this file